olp-cpp-sdk-dataservice-read/src/repositories/CatalogRepository.h

```diff
@@ -38,20 +39,21 @@ namespace repository {
 
 class CatalogRepository final {
  public:
-  static CatalogResponse GetCatalog(client::HRN catalog,
+  static CatalogResponse GetCatalog(client::ApiLookupClient client,
```

Comment:
I don't like this approach. Better move away from static methods and add CatalogClient constructor which takes in HRN and ApiLookupClient.
Same for all other repository classes.
On the long run we will have more and more parameters.